src/redux/reviews-moderation/reviews-moderation-api.ts

```diff
@@ -1,18 +1,81 @@
 import { HttpMethods } from "~/common/enums/index.ts";
-import { type GetModerationReviews } from "~/common/types/index.ts";
+import { type GetModerationReviewsResponse } from "~/common/types/index.ts";
+import { ModerationReviews } from "~/common/types/review/get-moderation-reviews.ts";
 
 import { api } from "../services.ts";
 import { reviewsApiPath } from "./constans.ts";
 
+type ModerationReviewsRequest = {
+	id?: string;
+	ordering?: "-time_added" | "time_added";
+	status?: "approved" | "pending" | "rejected";
+	type?: "company" | "course";
+};
+
 export const reviewsModerationApi = api.injectEndpoints({
 	endpoints: (builder) => ({
-		getReviewsModeration: builder.query<GetModerationReviews, void>({
+		getReviewsModeration: builder.query<GetModerationReviewsResponse, void>({
 			query: () => ({
 				method: HttpMethods.GET,
 				url: reviewsApiPath.ROOT,
 			}),
 		}),
+		getReviewsModerationByFilter: builder.query<
+			GetModerationReviewsResponse,
+			ModerationReviewsRequest
+		>({
+			forceRefetch({ currentArg, previousArg }) {
+				return (
+					currentArg?.type !== previousArg?.type ||
+					currentArg?.status !== previousArg?.status ||
+					currentArg?.ordering !== previousArg?.ordering ||
+					currentArg?.id !== previousArg?.id
+				);
+			},
+			query: (filters: ModerationReviewsRequest = {}) => {
+				return {
+					method: HttpMethods.GET,
+					params: {
+						ordering: filters.ordering,
+						status: filters.status,
+						type: filters.type,
+					},
```

Comment:
why don't param: filters?

Reply:
becouse param filters has more key than this query need